### PR TITLE
feat: use youtube controls in videojs player

### LINKED ISSUE #1103: Use YouTube player controls when playing back a YouTube video
We should use the standard YouTube video controls when playing back YouTube videos. 

See https://github.com/mitodl/hq/discussions/418 for a full discussion of the reasoning behind this decision. 

#### Designs and Mockups

Switching to the YouTube controls means that we have to move the download button outside of the video player

See these InVision designs for the new location and behavior:
- https://projects.invisionapp.com/d/main?origin=v7#/console/21537330/472347461/preview
- https://projects.invisionapp.com/d/main?origin=v7#/console/21537330/472347462/preview

#### Acceptance Criteria:

- [x] When a video is streamed from YouTube, use the YouTube controller (note: all our videos stream from YouTube at this time) 
- [x] While it is rare, it is possible for a video to have no interactive captions. In that case, the "Transcript" button should not be displayed, but the download button should still be available. 
- [x] remove the extra space to the left of the interactive transcript, so it left aligns with the video player (see [InVision Comment](https://projects.invisionapp.com/d/#/console/21537330/472347462/comments/150602733))
- [ ] YouTube controller and the design changes should also work for embedded videos (eg., see this [embedded video page](https://ocw.mit.edu/courses/res-18-009-learn-differential-equations-up-close-with-gilbert-strang-and-cleve-moler-fall-2015/pages/video-series-overview/))

#### Related issues

- Switching to the YouTube controller should also close #978

### PR DESCRIPTION
**Pull request by ibrahimjaved12**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
- [x] Testing
  - [ ] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #1103 

#### What's this PR do?
This PR:

- Replaces our custom defined videoJs controller with YouTube controller (which enables YouTube keyboard shortcuts) for online courses.
- Applies the styling for video tabs (eg. Transcript, Related Resources) as shown in [link1](https://projects.invisionapp.com/d/main?origin=v7#/console/21537330/472347461/preview) and [link2](https://projects.invisionapp.com/d/main?origin=v7#/console/21537330/472347462/preview)
- Removes the extra space to the left of the interactive transcript, so it left aligns with the video player (see [InVision Comment](https://projects.invisionapp.com/d/#/console/21537330/472347462/comments/150602733))

#### How should this be manually tested?
This will have to be tested on such courses: **a)** Course that has interactive captions  **b)** Course that does not have interactive captions **c)** Course with Related resources **d)** Course with Lecture Notes
Reason for b) is that we want to verify that video tab bar still appears with the download popup.
The other a), c), d) are for verifying styling changes.
**Sample courses for testing:** 
1) [18.06-spring-2010](https://github.mit.edu/ocw-content-rc/18.06-spring-2010) (for interactive captions + related resources + download button)
2) [3.091-fall-2018](https://github.mit.edu/ocw-content-rc/3.091-fall-2018) (for only download button)
3) [15.s21-january-iap-2014](https://github.mit.edu/ocw-content-rc/15.s21-january-iap-2014) (for lecture notes)

For styling changes, check for mobile responsiveness as well.
**Steps to test:**

1. Clear your cache.
2. Use incognito on browser
3. Checkout to this branch.
Now repeat these steps for each course. (cache does not have to be cleared again)
4. `OCW_TEST_COURSE=course_id`
5. `yarn start course`
6. Go to http://localhost:3000/ and go to Video Lectures on the left panel. Open any video.
7. Verify that YouTube controller is playing the video as opposed to videoJs custom controller.
8. Make sure after you press Play, the keyboard shortcuts are functional (eg. F for full screen, arrow keys, c key for caption, etc)
9. Verify that the design/styling changes are according to InVision designs:  [link1](https://projects.invisionapp.com/d/main?origin=v7#/console/21537330/472347461/preview) and [link2](https://projects.invisionapp.com/d/main?origin=v7#/console/21537330/472347462/preview)

For testing non-interactive captions `3.091-fall-2018` course, verify that "Transcript" text does not appear but the download icon appears (see screenshots below). But to test it we will have to make some changes in our code, that's the only way to test it locally for now:

In the file: base-theme/layouts/partials/video.html
line 25-29:
`	{{ if $captionsLocation }}
		{{ partial "video_expandable_tab.html" (dict "tabClass" "transcript" "tabTitle" "Transcript" "downloadLink" $downloadLink "transcriptLink" $transcriptPdfLocation) }}
	{{ else }}
		{{ partial "video_expandable_tab.html" (dict "tabClass" "" "tabTitle" "" "downloadLink" $downloadLink "transcriptLink" $transcriptPdfLocation) }}
	{{end}}`

Replace this code with:
`	{{ if $captionsLocation }}
		{{ partial "video_expandable_tab.html" (dict "tabClass" "" "tabTitle" "" "downloadLink" $downloadLink "transcriptLink" $transcriptPdfLocation) }}
	{{end}}`

Save. This time do clear your cache, and run on incognito. Run `yarn start course`.  Make sure that the transcript button does not appear on video but the download icon still appears.

#### Any background context you want to provide?
It will be helpful for the reviewer to read this full discussion before testing this PR: https://github.com/mitodl/hq/discussions/418#discussion-4765581

For code review, these points are important for the design changes. The previous custom videoJs controller is not removed, but it is disabled by CSS like `display: none`. The reason for this is we may want to use it for offline courses sometime in the future.

### For desktop:
image
image
image
image
image
image
image

### For mobile:
image

### CODE REVIEW

**Comment by ibrahimjaved12** [comment]
@pdpinch Could you please look at the screenshots and confirm the overall look of the styling is okay. Because I had to make some assumptions:

1. The transcript text left gap for mobile screens (For reference the last screenshot on the post)
2. 
> "While it is rare, it is possible for a video to have no interactive captions. In that case, the "Transcript" button should not be displayed, but the download button should still be available."

For this, if the interactive captions are not available, the download icon will have a standalone bar even if another bar is available (eg. for related resources or lecture notes). The reason for this is I thought that download icon (for transcript, and video) would suit best to transcript bar (and not for related resources, or lecture notes)
(For reference second last screenshot on the post)

**Comment by ChristopherChudzicki** [comment]
> For this, if the interactive captions are not available, the download icon will have a standalone bar even if another bar is available (eg. for related resources or lecture notes).

That makes sense to me.

**Comment by ibrahimjaved12** [comment]
Thanks @ChristopherChudzicki 

Waiting for @pdpinch's confirmation to this comment https://github.com/mitodl/ocw-hugo-themes/pull/1115#issuecomment-1488967422
And then the PR can be reviewed

**Comment by pdpinch** [comment]
> 1. The transcript text left gap for mobile screens ([Click me for screenshot](https://user-images.githubusercontent.com/109785089/228611105-b5d989ea-8bea-400d-9f7c-cd431f63a579.png))

This looks OK to me as is. But if it's easier to make the left alignments the same on desktop as they are on mobile, I would be OK with that too. 

> 2. if the interactive captions are not available, the download icon will have a standalone bar even if another bar is available (eg. for related resources or lecture notes). 

This is OK for now. It's not great, but I don't have a better solution. 

> 3. The hover on the transcripts ([Click me for screenshot](https://user-images.githubusercontent.com/109785089/228608792-a425a7cf-31ac-424c-98f9-486585cd0133.png))

This looks good to me. Is it different from the current hover?

**Comment by ibrahimjaved12** [comment]
Thanks @pdpinch .


> This looks OK to me as is. But if it's easier to make the left alignments the same on desktop as they are on mobile, I would be OK with that too.

Done

> This looks good to me. Is it different from the current hover?

Yes, there are minor color differences in the hovers:
image
For the new one, I just matched it with the active line color.

**Comment by ibrahimjaved12** [comment]
Updated PR with new mobile changes (left gap on transcript now matches the left gap on desktop):
image

File: course-v2/assets/videojs-imports.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { initVideoTranscriptTrack } from "./js/video_transcript_track"
 import { initPlayBackSpeedButton } from "./js/video_playback_speed"
 import { initVideoFullscreenToggle } from "./js/video_fullscreen_toggle"
 import { initDownloadButton } from "./js/video-download-button"
+import { videoToggleDownloadPopup } from "./js/video_toggle_download"
```
Comment (HussainTaj-arbisoft) [inline course-v2/assets/videojs-imports.js:5]: Following the existing convention, I think it'd be nice to name this method something like `initVideoDownloadPopup`.

Comment (ibrahimjaved12) [reply course-v2/assets/videojs-imports.js]: Thanks for your suggestion. I have updated the function name along with the file name.

**Comment by pdpinch** [comment]
Sorry, I should have asked before this was merged: will this change to the player controls apply to the embedded video player as well as the single page video player? It should. 

See for example, https://ocw.mit.edu/courses/res-18-009-learn-differential-equations-up-close-with-gilbert-strang-and-cleve-moler-fall-2015/resources/gilbert-and-cleve-introduction/

**Comment by ibrahimjaved12** [comment]
> Sorry, I should have asked before this was merged: will this change to the player controls apply to the embedded video player as well as the single page video player? It should.
> 
> See for example, https://ocw.mit.edu/courses/res-18-009-learn-differential-equations-up-close-with-gilbert-strang-and-cleve-moler-fall-2015/resources/gilbert-and-cleve-introduction/

Yes, it applies to embedded video player as well. 
The change can be checked from live-qa for the same video you mentioned: https://live-qa.ocw.mit.edu/courses/res-18-009-learn-differential-equations-up-close-with-gilbert-strang-and-cleve-moler-fall-2015/resources/gilbert-and-cleve-introduction/

**Pull request merged**